airflow/jobs.py

```diff
@@ -403,39 +422,77 @@ def schedule_dag(self, dag):
                     dr.end_date = datetime.now()
             session.commit()
 
+            # check last scheduled run
+            # get this dag's max execution date by
+            #  external trigger is false
+            #  and "like" Dag Run ID.ID
+            # bolke: why a like?
             qry = session.query(func.max(DagRun.execution_date)).filter_by(
-                    dag_id = dag.dag_id).filter(
-                        or_(DagRun.external_trigger == False,
-                            # add % as a wildcard for the like query
-                            DagRun.run_id.like(DagRun.ID_PREFIX+'%')))
+                dag_id=dag.dag_id).filter(
+                or_(DagRun.external_trigger is False,
+                    # add % as a wildcard for the like query
+                    DagRun.run_id.like(DagRun.ID_PREFIX+'%')
+                    )
+            )
             last_scheduled_run = qry.scalar()
+            self.logger.debug("Last scheduled run: {}".format(last_scheduled_run))
+
+            # Determine next run date
             next_run_date = None
             if dag.schedule_interval == '@once' and not last_scheduled_run:
                 next_run_date = datetime.now()
             elif not last_scheduled_run:
                 # First run
+                self.logger.debug("Scheduling {} for the first time".format(dag.dag_id))
+
+                # Get the maximum execution date for the tasks associated with this dag
                 TI = models.TaskInstance
                 latest_run = (
                     session.query(func.max(TI.execution_date))
                     .filter_by(dag_id=dag.dag_id)
                     .scalar()
                 )
+
                 if latest_run:
                     # Migrating from previous version
                     # make the past 5 runs active
+                    # todo: remove this?
```

Comment:
@mistercrunch Is this line still needed? I suspect this was needed to handle some airbnb migration. Can you add some comments as to what this (`next_run_date = dag.date_range(latest_run, -5)[0]`) is?